cmd/skuba/cluster/check.go

```diff
@@ -0,0 +1,133 @@
+}
+
+var (
+	format    string
+	filename  string
+	inputFile string
+)
+
+// newUpgradeCheckCmd creates a new `skuba check` cobra command
+func newUpgradeCheckCmd() *cobra.Command {
+	checkOptions := &checkOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "check kubernetes-version=<version> swaggerDir=<directory> --api-walk=<true|fasle>",
+		Short: "Print Upgrade Check information",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubernetesConfigFlags := genericclioptions.NewConfigFlags(true)
+			kubeAdminFile := skuba.KubeConfigAdminFile()
+			kubernetesConfigFlags.KubeConfig = &kubeAdminFile
+
+			config := lib.Config{
+				K8sVersion:      checkOptions.K8sVersion,
+				ForceDownload:   checkOptions.ForceDownload,
+				APIWalk:         checkOptions.APIWalk,
+				SwaggerDir:      checkOptions.SwaggerDir,
+				ShowDescription: checkOptions.ShowDescription,
+				ConfigFlags:     kubernetesConfigFlags,
+				Input:           inputFile,
+			}
+
+			if config.K8sVersion == "" {
+				client, err := kubernetes.GetAdminClientSet()
+				if err != nil {
+					return err
+				}
+				currentClusterVersion, err := kubeadm.GetCurrentClusterVersion(client)
+				if err != nil {
+					return err
+				}
+				availableVersions := kubernetes.AvailableVersions()
+				upgradePath, err := upgrade.CalculateUpgradePath(currentClusterVersion, availableVersions)
+				if err != nil {
+					return err
+				}
+
+				var nextClusterVersion *version.Version
+				if len(upgradePath) > 0 {
+					nextClusterVersion = upgradePath[0]
+				} else {
+					klog.Warning("Already on the latest version, nothing to check.\nFor a specific version use the `kubernetes-version` flag.")
+					return nil
+				}
+				config.K8sVersion = fmt.Sprintf("v%s", nextClusterVersion.String())
+			}
+
+			klog.Infof("Starting Kubepug with configs: %+v", config)
+			kubepug := lib.NewKubepug(config)
+
+			result, err := kubepug.GetDeprecated()
+			if err != nil {
+				return err
+			}
+
+			klog.Info("Starting deprecated objects printing")
+			formatter := formatter.NewFormatter(format)
+			bytes, err := formatter.Output(*result)
+			if err != nil {
+				return err
+			}
+
+			if filename != "" {
+				err = ioutil.WriteFile(filename, bytes, 0644)
+				if err != nil {
+					return err
+				}
+			} else {
+				fmt.Printf("%s", string(bytes))
+			}
+
+			return nil
+		},
+	}
+	cmd.PersistentFlags().BoolVar(&checkOptions.APIWalk, "api-walk", false, "Whether to walk in the whole API, checking if all objects type still exists in the current swagger.json. May be IO intensive to APIServer.")
+	cmd.PersistentFlags().BoolVar(&checkOptions.ShowDescription, "description", true, "Wether to show the description of the deprecated object. The description may contain the solution for the deprecation.")
+	cmd.PersistentFlags().StringVar(&checkOptions.K8sVersion, "kubernetes-version", "", "Which kubernetes release version (https://github.com/kubernetes/kubernetes/releases) should be used to validate objects.")
```

Comment:
If we support to let admin/user to specifies different Kubernetes version, I _guess_ the kubepug would need to access external Kubernetes swagger.json resource.

Then, will it fail when the admin/user in the air-gapped environment?

Reply:
That could happen, but you can provide a `SwaggerDir` option, and if you have the file there it will not go and fetch the file online, it will use the one in that folder.

Reply:
I would suggest having some documentation in `doc-caasp` for the air-gapped case.

Reply:
Here is the PR, still pending review from the Docs team
https://github.com/SUSE/doc-caasp/pull/916